src/sdk/index.js

```diff
@@ -4,23 +4,46 @@ import {merge} from 'lodash'
 import api from './api'
 import {DEFAULT_HEADERS} from './headers'
 
-export default function sdk (globalSpec = {}) {
+function configGlobalSpec (globalSpec) {
+
+  let headers
+  if (typeof globalSpec.headers === 'function') {
+    headers = globalSpec.headers()
+  }
+  else {
+    headers = globalSpec.headers
+  }
+
+  return {
+    ...globalSpec,
+    headers
+  }
+}
 
-  let defaultSpec = {
+function configOptions (globalSpec, instanceSpec = {}) {
+
+  const defaultSpec = {
     origin: '',
     headers: DEFAULT_HEADERS
   }
-  let options = merge({}, defaultSpec, globalSpec)
+
+  return merge({}, defaultSpec, configGlobalSpec(globalSpec), instanceSpec)
```

Comment:
if you added `globalSpec` right after `defaultSpec` then you wouldn't have to spread globalSpec on line 18